### PR TITLE
Mangroves plugin

### PR DESCRIPTION
**Pull request by emmaai**
Add mangroves plugin

### CODE REVIEW

File: odc/stats/plugins/mangroves.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+"""
+Fractional Cover Percentiles
+"""
```
Comment (SpacemanPaul) [inline odc/stats/plugins/mangroves.py:3]: File docstring needs to be updated.

File: odc/stats/plugins/mangroves.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        gdal.RasterizeLayer(target_ds, [1], source_layer, burn_values=[1])
+        return dask.array.from_array(band.ReadAsArray().reshape(array_shape), name=False)
+
+    def native_transform(self, xx):
+        """
+            no native transform required since
+            no reprojection in mangroves
+            return loaded data
+        """
+        return xx
```
Comment (SpacemanPaul) [inline odc/stats/plugins/mangroves.py]: This is OK, but unnecessary as it is identical to the default implementation is provided by the base class.

File: odc/stats/plugins/mangroves.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    def fuser(self, xx):
+        """
+            no fuse required for mangroves either
+            return loaded data
+        """
+        return xx
```
Comment (SpacemanPaul) [inline odc/stats/plugins/mangroves.py:74]: This is different to the default implementation.  Are you sure this is what you want?

**Review by SpacemanPaul — CHANGES_REQUESTED** [review]
At the very least please change the file docstring to say "Mangroves" instead of "Fractional Cover".

Also please double check whether the fuser function is correct or if you actually want the _nodata_fuser() default implementation.   It might be unnecessary because you are doing masking in the reduce stage instead of the dataload stage.

Nice to see some tests though!

File: odc/stats/plugins/mangroves.py (hunk 3 above)
Comment (emmaai) [reply odc/stats/plugins/mangroves.py:74]: Yes, I'm sure. There is nothing to group, hence nothing to fuse.

File: odc/stats/plugins/mangroves.py (hunk 2 above)
Comment (emmaai) [reply odc/stats/plugins/mangroves.py]: deleted

**Review by SpacemanPaul — APPROVED** [review]

**Pull request merged**